src/test/java/com/fasterxml/jackson/core/json/async/AsyncStringArrayTest.java

```diff
@@ -128,6 +128,34 @@ public void testLongUnicodeStrings() throws IOException
         _testStrings(f, input, data, 1, 1);
     }
 
+    public void testLongAsciiStringsSmallLimit() throws IOException
```

Comment:
@cowtowncoder the test I added to passes when it is run by itself but fails when it is run with other tests. It seems that running other tests with the default StreamReadConstraints seems to leave behind state - maybe in the BufferRecyclers - that means the validateStringLength checks don't get called.

Reply:
Yes, due to buffer recycling it is possible that recycled buffer has reached size of up to 64k chars.

So test will probably be set as much less strict: to set maximum length to say 100k, produce 200k String and verify that gets caught.

Another thing we could consider: making `MAX_SEGMENT_LEN` no higher than `maxStringLength`.
Then again... maybe not worth bothering yet.